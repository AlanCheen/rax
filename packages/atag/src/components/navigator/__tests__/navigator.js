describe('navigator', () => {
  const pages = [];
  async function createPage() {
    const page = await global.__BROWSER__.newPage();
    await page.setViewport(global.__VIEWPORT__);
    await page.goto('http://localhost:9002/components/navigator/__tests__/navigator.html');
    pages.push(page);
    return page;
  }

  afterAll(async() => {
    for (let i = 0, l = pages.length; i < l; i ++) {
      await pages[i].close();
    }
  });

  it('url', async() => {
    const page = await createPage();
    const url = 'foo';
    await page.addScriptTag({
      content: `
        const navigator = document.querySelector('#navigator');
        navigator.url = '${url}';
      `,
    });
    await page.click('#navigator');
    expect(await page.evaluate('__renderer_to_worker_message__')).toMatchSnapshot();
  });

  it('open-type', async() => {
    const page = await createPage();
    const url = 'foo';
    await page.addScriptTag({
      content: `
        const navigator = document.querySelector('#navigator');
        navigator.url = '${url}';
        navigator.openType = 'redirect';
      `,
    });
    await page.click('#navigator');
    expect(await page.evaluate('__renderer_to_worker_message__')).toMatchSnapshot();
  });
});
