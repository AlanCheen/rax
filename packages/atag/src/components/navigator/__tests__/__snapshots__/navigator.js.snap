// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigator open-type 1`] = `
Object {
  "navigateTo": "foo",
  "navigateType": "redirect",
  "type": "navigate",
}
`;

exports[`navigator url 1`] = `
Object {
  "navigateTo": "foo",
  "navigateType": "navigate",
  "type": "navigate",
}
`;
